iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/NodeBreakerVoltageLevel.java

```diff
@@ -528,18 +528,14 @@ public void edgeRemoved(int e, SwitchImpl aSwitch) {
             }
 
             @Override
-            public void allEdgesRemoved() {
-                List<String> removedSwitchesIds = new ArrayList<>(graph.getEdgeCount());
+            public void allEdgesRemoved(Collection<SwitchImpl> aSwitches) {
                 NetworkImpl network = getNetwork();
-                for (SwitchImpl s : graph.getEdgesObject()) {
-                    if (s != null) {
-                        network.getListeners().notifyBeforeRemoval(s);
-                        removedSwitchesIds.add(s.getId());
-                        network.getIndex().remove(s);
-                    }
-                }
+                aSwitches.forEach(ss -> {
```

Comment:
Is there any reason to have a difference between the `aSwitches` passed on and the `switches` map?

Reply:
`aSwitches` contains the list of switches while `switches` contains ids and the associated edge's number